### PR TITLE
Pr/1297

### PR DESCRIPTION
**Pull request by megawac**
/cc @CodeMan99 we decided to avoid importing lodash/filter in order to avoid including a bunch of unused deep property handling methods (that let lodash support `_.filter(arr, 'a.b.c')`) in our distribution files

Fixes #1297

### CODE REVIEW

File: lib/internal/filter.js [hunk 1]
```diff
@@ -1,31 +1,27 @@
...
-        }
-        else {
-            callback(null, arrayMap(results.sort(function (a, b) {
-                return a.index - b.index;
-            }), property('value')));
+        if (err) return callback(err);
+        var result = [];
+        var nextElem = iterator(coll);
+        var elem;
+        while ((elem = nextElem()) !== null) {
```
Comment (megawac) [inline lib/internal/filter.js]: So there is a drawback/regression for doing it this way:

When `coll` is an iterator we will be trying to instantiate the iterator twice; sometimes an iterator can only be run once which would cause us to return an empty list.

**Comment by megawac** [comment]
I've decided to revert the implementation change due to edge case handling (iterating iterables and objects) but will merge this for the test cases

**Comment by CodeMan99** [comment]
Woo! Test cases.

**Comment by megawac** [comment]
Thanks for trying to get this refactor in @CodeMan99 sorry it didn't work out. Regardless these test cases should help for the future

**Comment by CodeMan99** [comment]
Yeah, it was one of those cases where I wrote a solution with with `async.map` and wondered if I could reduce the number loops with another tool. At which point I landed on `async.filter` but was pretty confused when I read the source and saw the sorting. Then I set off trying to reimplement to make it faster. You know the rest of the story from there.

At any rate, it was nice working with you. 🎩
